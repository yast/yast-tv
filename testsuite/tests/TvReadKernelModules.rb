# encoding: utf-8

module Yast
  class TvReadKernelModulesClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      @READ_modules = {
        "modinfo" => {
          "kernel" => {
            "drivers" => {
              "media" => {
                "video" => {
                  "bttv"  => {
                    "module_filename"    => "/lib/modules/2.4.4-4GB/kernel/drivers/media/video/bttv.o",
                    "module_description" => "bttv - v4l driver module for bt848/878 based cards",
                    "tuner"              => "specify installed tuner type"
                  },
                  "zoran" => {
                    "module_description" => "Zoran ZR36120 based framegrabber",
                    "module_filename"    => "/lib/modules/2.4.4-4GB/kernel/drivers/media/video/zoran.o"
                  }
                },
                "radio" => {
                  "miropcm20" => {
                    "module_author"      => "Ruurd Reitsma",
                    "module_description" => "A driver for the Miro PCM20 radio card.",
                    "module_filename"    => "/lib/modules/2.4.19-4GB/kernel/drivers/media/radio/miropcm20.o",
                    "radio_nr"           => ""
                  }
                }
              }
            }
          }
        }
      }

      # none kernel modules
      TEST(lambda { Tv.ReadKernelModules }, [{}, {}, {}], nil)
      DUMP(Tv.kernel_modules)
      DUMP(Tv.radio_modules)

      Tv.kernel_modules = nil #enforce new read

      TEST(lambda { Tv.ReadKernelModules }, [@READ_modules, {}, {}], nil)
      DUMP(Tv.kernel_modules)
      DUMP(Tv.radio_modules)

      nil
    end
  end
end

Yast::TvReadKernelModulesClient.new.main
