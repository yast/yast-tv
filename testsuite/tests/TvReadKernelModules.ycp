{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    map READ_modules =
	$[ "modinfo" :
	    $[ "kernel" :
		$[ "drivers" :
		    $["media" :
			$[ "video" :
			    $[ "bttv" : $["module_filename" : "/lib/modules/2.4.4-4GB/kernel/drivers/media/video/bttv.o",
				    "module_description" : "bttv - v4l driver module for bt848/878 based cards",
				    "tuner":"specify installed tuner type"],
				"zoran" : $["module_description" : "Zoran ZR36120 based framegrabber",
				    "module_filename" : "/lib/modules/2.4.4-4GB/kernel/drivers/media/video/zoran.o"]],
               "radio":
                $["miropcm20":$["module_author":"Ruurd Reitsma",
                    "module_description":"A driver for the Miro PCM20 radio card.",
                    "module_filename":"/lib/modules/2.4.19-4GB/kernel/drivers/media/radio/miropcm20.o",
                    "radio_nr":""]]
    ]]]]];

    // none kernel modules
    TEST (``(Tv::ReadKernelModules()), [ $[], $[], $[] ], nil);
    DUMP (Tv::kernel_modules);
    DUMP (Tv::radio_modules);

    TEST (``(Tv::ReadKernelModules()), [ READ_modules, $[], $[] ], nil);
    DUMP (Tv::kernel_modules);
    DUMP (Tv::radio_modules);
}
