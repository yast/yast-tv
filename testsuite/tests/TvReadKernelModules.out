Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.Locale nil
Read	.etc.install_inf.Instmode nil
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dir	.modinfo.kernel.drivers.media.video: []
Return	true
Dump	$[]
Dir	.modinfo.kernel.drivers.media.video: ["bttv", "zoran"]
Read	.modinfo.kernel.drivers.media.video."bttv" $["module_description":"bttv - v4l driver module for bt848/878 based cards", "module_filename":"/lib/modules/2.4.4-4GB/kernel/drivers/media/video/bttv.o", "tuner":"specify installed tuner type"]
Read	.modinfo.kernel.drivers.media.video."zoran" $["module_description":"Zoran ZR36120 based framegrabber", "module_filename":"/lib/modules/2.4.4-4GB/kernel/drivers/media/video/zoran.o"]
Return	true
Dump	$["bttv":$["module_description":"bttv - v4l driver module for bt848/878 based cards", "module_filename":"/lib/modules/2.4.4-4GB/kernel/drivers/media/video/bttv.o", "tuner":"specify installed tuner type"], "zoran":$["module_description":"Zoran ZR36120 based framegrabber", "module_filename":"/lib/modules/2.4.4-4GB/kernel/drivers/media/video/zoran.o"]]
